jsonapi/unmarshal.go

```diff
@@ -285,6 +285,13 @@ func UnmarshalInto(input map[string]interface{}, targetStructType reflect.Type,
 							}
 
 							field.Set(reflect.ValueOf(t))
+						case *time.Time:
```

Comment:
ah yes this makes sense, `time.Time` was also treated as a special case so it must be the same with the pointer type. We really have to start with refactoring #111